forte/data/data_pack_dataset.py

```diff
@@ -0,0 +1,278 @@
+                 request: Optional[DataRequest] = None,
+                 skip_k: int = 0):
+        self._iterator: Iterator = DataPackIterator(pack_generator,
+                                                    context_type,
+                                                    request,
+                                                    skip_k)
+        super().__init__(self)
+
+    def __iter__(self):
+        return self._iterator
+
+
+class DataPackDataset(DatasetBase):
+    """
+    A dataset representing data packs. Calling an
+    `DataIterator
+    <https://texar-pytorch.readthedocs.io/en/latest/code/data.html#dataiterator>`_
+    over this `DataPackDataset` will produce an `Iterate` over batch of examples
+    parsed by a reader from given data packs.
+
+    Args:
+        data_source: A data source of type
+            :class:`forte.data.data_pack_dataset.DataPackDataSource`.
+        feature_schemes (dict): A `Dict` containing all the information to do
+            data pre-processing. This is exactly the same as the `schemes` in
+            `feature_resource`. Please refer to :meth:`feature_resource` in
+            :class:`forte.train_preprocessor.TrainPreprocessor` for details.
+        hparams: A `dict` or instance of :
+            class:`forte.common.configuration.Config` containing
+            hyperparameters. See :meth:`default_hparams` in
+            `DatasetBase
+            <https://texar-pytorch.readthedocs.io/en/latest/code/data.html#datasetbase>`
+            for the defaults.
+        device: The device of the produced batches. For GPU training,
+            set to current CUDA device.
+    """
+    def __init__(self,
+                 data_source: DataPackDataSource,
+                 feature_schemes: Dict,
+                 hparams: Union[Dict, HParams] = None,
+                 device: Optional[torch.device] = None):
+        self._data_source = data_source
+        self._feature_scheme = feature_schemes
+
+        super().__init__(self._data_source, hparams, device)
+
+    def process(self, raw_example: RawExample) -> FeatureCollection:
+        """
+        Given an input which is a single data example, extract feature from it.
+
+        Args:
+            raw_example (tuple(dict, DataPack)): A `Tuple` where
+
+                The first element is a `Dict` produced by :meth:`get_data()` in
+                :class:`forte.data.data_pack.DataPack`.
+
+                The second element is an instance of type
+                :class:`forte.data.data_pack.DataPack`.
+
+        Returns:
+            A `Dict` mapping from user-specified tags to the
+            :class:`forte.data.converter.Feature` extracted.
+
+            .. note::
+                Please refer to Please refer to :meth:`feature_resource` in
+                :class:`forte.train_preprocessor.TrainPreprocessor` for details
+                about user-specified tags.
+        """
+        instance: Instance = raw_example[0]
```

Comment:
It looks like only `tid` is used in this instance. Anything else is useful from the instance?

Reply:
Currently no. I will change the instance to be only`tid`